### PR TITLE
More unit-test cases for #459

### PR DESCRIPTION
**Pull request by johnkerl**
Follow-up from
https://github.com/single-cell-data/TileDB-SOMA/pull/459#pullrequestreview-1153616376

### CODE REVIEW

File: apis/python/tests/test_soma_dataframe.py [hunk 1]
```diff
@@ -238,7 +238,11 @@ def _check_tbl(tbl, col_names, ids, *, demote):
 def test_empty_soma_dataframe(tmp_path):
     a = soma.SOMADataFrame((tmp_path / "A").as_posix())
     a.create(pa.schema([("a", pa.bool_())]))
-    next(a.read())
+    # Must not throw
+    assert len(next(a.read())) == 0
+    assert len(a.read_all()) == 0
+    assert len(next(a.read_as_pandas())) == 0
+    assert len(a.read_as_pandas_all()) == 0
```
Comment (bkmartinjr) [inline apis/python/tests/test_soma_dataframe.py:245]: Given the bug fix, we also need:
```
assert isinstance(a.read_as_pandas_all(), pd.DataFrame)
```

File: apis/python/tests/test_soma_indexed_dataframe.py [hunk 2]
```diff
@@ -186,7 +186,11 @@ def _check_tbl(tbl, col_names, ids, *, demote):
 def test_empty_soma_indexed_dataframe(tmp_path):
     a = soma.SOMAIndexedDataFrame((tmp_path / "A").as_posix())
     a.create(pa.schema([("a", pa.int32())]), index_column_names=["a"])
-    next(a.read())
+    # Must not throw
+    assert len(next(a.read())) == 0
+    assert len(a.read_all()) == 0
+    assert len(next(a.read_as_pandas())) == 0
+    assert len(a.read_as_pandas_all()) == 0
```
Comment (bkmartinjr) [inline apis/python/tests/test_soma_indexed_dataframe.py:193]: Given the bug fix, we also need:
```
assert isinstance(a.read_as_pandas_all(), pd.DataFrame)
```

**Review by bkmartinjr — APPROVED** [review]
One requested addition so that the test covers the bug fix.  Otherwise looks great.

**Comment by johnkerl** [comment]
> One requested addition so that the test covers the bug fix. Otherwise looks great.

@bkmartinjr commit #1 on this PR added a test, which failed as intended. Commit #2 fixed the fail. I sequenced these two commits that way by design.

**Pull request merged**